internal/controller/machineaccount_controller.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	iammiloapiscomv1alpha1 "go.miloapis.com/milo/pkg/apis/iam/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/finalizer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"go.miloapis.com/auth-provider-zitadel/internal/zitadel"
+)
+
+const (
+	machineAccountFinalizerKey = "iam.miloapis.com/machineaccount"
+)
+
+// MachineAccountReconciler reconciles a MachineAccount object
+type MachineAccountController struct {
+	Client     client.Client
+	Finalizers finalizer.Finalizers
+	Scheme     *runtime.Scheme
+	Zitadel    *zitadel.Client
+}
+
+type machineAccountFinalizer struct {
+	Client  client.Client
+	Zitadel *zitadel.Client
+}
+
+func (f *machineAccountFinalizer) Finalize(ctx context.Context, obj client.Object) (finalizer.Result, error) {
```

Comment:
Just noting that when you implement the MachineAccountKey functionality, you will want to make sure that keys are deleted before the MachineAccount is deleted.